src/auth/newuserwindow.cpp

```diff
@@ -0,0 +1,103 @@
+#include "newuserwindow.h"
+
+#include<QVBoxLayout>
+#include<QLabel>
+#include<QLineEdit>
+#include <QApplication>
+#include <QDesktopWidget>
+#include <QDialogButtonBox>
+#include <QPushButton>
+#include <QDebug>
+#include <QToolTip>
+
+NewUserWindow::NewUserWindow(QWidget *parent) : QDialog(parent)
+{
+    setModal(true);
+    window()->setFixedSize( window()->sizeHint() );
+    setWindowTitle("Create new account");
+    QVBoxLayout *mainLayout = new QVBoxLayout;
+    mainLayout->addStretch(mBasicStretch);
+
+    //  login parameters
+    QHBoxLayout *loginLayout = new QHBoxLayout;
+    mpLabelLogin = new QLabel("GitHub login", this);
+    mpLabelLogin->setMinimumWidth(mLabelWidth);
+    mpLabelLogin->setMaximumWidth(mLabelWidth);
+    mpEditLogin = new QLineEdit(this);
+    mpEditLogin->setMinimumWidth(mEditWidth);
+    mpEditLogin->setMaximumWidth(mEditWidth);
+    loginLayout->addWidget(mpLabelLogin);
+    loginLayout->addWidget(mpEditLogin);
+    mainLayout->addLayout(loginLayout);
+
+    //  token/password parameters
+    QHBoxLayout *tokenLayout = new QHBoxLayout;
+    mpLabelToken = new QLabel("Password or token", this);
+    mpLabelToken->setMinimumWidth(mLabelWidth);
+    mpLabelToken->setMaximumWidth(mLabelWidth);
+    mpEditToken = new QLineEdit(this);
+    mpEditToken->setMinimumWidth(mEditWidth);
+    mpEditToken->setMaximumWidth(mEditWidth);
+    mpEditToken->setEchoMode(QLineEdit::Password);
+    mpEditToken->setPlaceholderText(mPlaceholderText);
+    mpEditToken->setToolTip(mPlaceholderText);
+    tokenLayout->addWidget(mpLabelToken);
+    tokenLayout->addWidget(mpEditToken);
+    mainLayout->addLayout(tokenLayout);
+
+    // DialogButtonBox  parameters
+    mpButtonBox = new QDialogButtonBox(QDialogButtonBox::Ok | QDialogButtonBox::Cancel, this);
+    //mpButtonBox->button( QDialogButtonBox::Ok )->setEnabled( false );
+    connect(mpButtonBox, &QDialogButtonBox::clicked, this, &NewUserWindow::onBtnBoxClicked);
+    mainLayout->addWidget(mpButtonBox);
+
+    // separator
+    QFrame *line = new QFrame(this);
+    line->setFrameShape(QFrame::HLine);
+    line->setFrameShadow(QFrame::Sunken);
+    line->setLineWidth(mSeparatorWidth);
+    mainLayout->addWidget(line);
+
+    // unnamed user button
+    QHBoxLayout *unnamedButtonLayout = new QHBoxLayout;
+    mpUnnamedUserButton = new QPushButton("Try without authorizattion", this);
+    mpUnnamedUserButton->setMinimumWidth(mUnnamedUserButtonWidth);
+    mpUnnamedUserButton->setMaximumWidth(mUnnamedUserButtonWidth);
+    unnamedButtonLayout->addWidget(mpUnnamedUserButton);
+    mainLayout->addLayout(unnamedButtonLayout);
+    connect(mpUnnamedUserButton, &QPushButton::clicked, this, [=](){emit unnamedUser();});
+    setLayout(mainLayout);
+    show();
+    move((QApplication::desktop()->width() - this->width()) / 2 ,
+         (QApplication::desktop()->height() - this->height()) / 2);
+}
```

Comment:
```
show();
 move((QApplication::desktop()->width() - this->width()) / 2 ,
 (QApplication::desktop()->height() - this->height()) / 2);
```
you already have used the same code in choiseWindow.cpp. Create a fucntion for setting widget geometry